### ISSUE TITLE
Using vitool to generate an AnimatedIcon

### ISSUE THREAD

**Issue by zencubic** [flutter/flutter#27858]
I have an animated SVG which I want to convert to a Flutter-usable AnimatedIcon. But I'm not looking through what format vitool needs. It would be great if vitool had a documentation. Also it would be helpful if we get example input files.

**Comment by zoechi** [comment]
https://github.com/flutter/flutter/tree/master/dev/tools/vitool

> is not intended to be a general-purpose tool.

You can ask the community for support.

Perhaps https://pub.dartlang.org/packages?q=flottie fits your need better anyway.

**Comment by zencubic** [comment]
@zoechi But both lottie packages don't have support for iOS and I think AnimatedIcons could be great and easy to use if there were a bit documentation. Also you don't need extra libraries in your App for maybe just a single button. So I don't see the point of using lottie for this

**Comment by zoechi** [comment]
> lottie packages don't have support for iOS

@dnfield can you confirm?

**Comment by zoechi** [comment]
@zencubic probably depends on #1831

**Comment by zencubic** [comment]
@zoechi [lottie_flutter](https://pub.dartlang.org/packages/lottie_flutter) is based on [lottie-android](https://github.com/airbnb/lottie-android) and [fluttie](https://pub.dartlang.org/packages/fluttie) explicitly states, that they have no iOS implementation

**Comment by dnfield** [comment]
There are two lottie packages for Flutter - one that uses the native Android Lottie plugin (fluttie), and one that uses Dart code to render it.  The Dart one (lottie_flutter) supports iOS and Android.

That said, it's not in the best shape and needs some rework.  

I still have it in my head to, at some point, look at what it would mean for us to make Skottie available as a module.  That really seems like the best solution, but has some technical challenges to figure out.  See https://github.com/flutter/flutter/issues/20438.  I would not sit around waiting for that one though, unfortunately - I'd like to think that at some point this year it will get some more attention, but no real guarantees.

**Comment by dnfield** [comment]
And just to clarify, even though lottie_flutter is based on the Android code, it's not actually using any Android code.  The original author (and then I) used the Android implementation as a reference point.

If Skottie doesn't pan out, I would like to completely rewrite lottie_flutter probably based on the web implementation (which is a little bit better than the Android one in terms of feature support).

**Comment by zencubic** [comment]
So for now it's only possible to use lottie_flutter for adding a custom animated icon into a flutter application?

**Comment by dnfield** [comment]
If I were building a Flutter app today and wanted to use a Lottie icon, I'd probably use lottie_flutter.

You could also poke around at flutter_svg - that does not support Animated SVG and it's kind of a non-goal for that project at this point.  Depending on the complexity of your SVG and what kind of animations, I might look at just converting it to Dart code and use an `AnimationController` to animate it.

**Comment by zencubic** [comment]
@dnfield Thank you for the clarification!
